src/zenml/zen_stores/migrations/versions/ccd68b7825ae_add_status_to_pipeline_and_step_run.py

```diff
@@ -0,0 +1,56 @@
+"""Add status to pipeline and step run [ccd68b7825ae].
+
+Revision ID: ccd68b7825ae
+Revises: c1b18cec3a48
+Create Date: 2022-10-24 16:49:37.007641
+
+"""
+import sqlalchemy as sa
+import sqlmodel
+from alembic import op
+
+# revision identifiers, used by Alembic.
+revision = "ccd68b7825ae"
+down_revision = "c1b18cec3a48"
+branch_labels = None
+depends_on = None
+
+
+def upgrade() -> None:
+    """Upgrade database schema and/or data, creating a new revision."""
+    with op.batch_alter_table("pipelinerunschema", schema=None) as batch_op:
+        batch_op.add_column(
+            sa.Column(
+                "status", sqlmodel.sql.sqltypes.AutoString(), nullable=True
+            ),
+        )
+    op.execute("UPDATE pipelinerunschema SET status = 'running'")
+    with op.batch_alter_table("pipelinerunschema", schema=None) as batch_op:
+        batch_op.alter_column(
+            "status",
+            nullable=False,
+            existing_type=sqlmodel.sql.sqltypes.AutoString(),
+        )
```

Comment:
You might be able to get away with a server_default instead of this three step approach  - https://www.loom.com/share/6b57aa48248245f9904de4ce1782a2e4

This will anchor the default value in the table definitions as well, not sure if that is wanted behaviour, that whenever the field is not specified explicitely it defaults to running. What do you think?

```suggestion
    with op.batch_alter_table("pipelinerunschema", schema=None) as batch_op:
        batch_op.add_column(
            sa.Column(
                "status", sqlmodel.sql.sqltypes.AutoString(), nullable=True, server_default="running"
            ),
        )
```

Reply:
@AlexejPenner , we explicitly tried to avoid using a server default, because that configures an additional layer of logic in the database that can create unpredictable behavior.